apis/beacon/pool/attestations.yaml

```diff
@@ -0,0 +1,54 @@
+get:
+  operationId: getPoolAttestations
+  summary: Get Attestations from operations pool
+  description: Retrieves attestations known by the node but not necessarily incorporated into any block
+  parameters:
+    - name: slot
+      in: query
+      required: false
+      schema:
+        $ref: '../../../beacon-node-oapi.yaml#/components/schemas/Uint64'
+    - name: committee_index
+      in: query
+      required: false
+      schema:
+        $ref: '../../../beacon-node-oapi.yaml#/components/schemas/Uint64'
+  tags:
+    - Beacon
+  responses:
+    "200":
+      description: Successful response
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              data:
+                type: array
+                items:
+                  $ref: '../../../beacon-node-oapi.yaml#/components/schemas/Attestation'
+    "500":
+      $ref: '../../../beacon-node-oapi.yaml#/components/responses/InternalError'
+
+post:
+  operationId: submitPoolAttestations
+  summary: Submit Attestation object to node
+  description: Submits Attestation object to node. If attestation passes all validation constraint, node MUST publish attestation on appropriate subnet.
+  tags:
+    - Beacon
+    - MinimalValidatorApi
+  requestBody:
+    required: true
+    content:
+      application/json:
+        schema:
+          $ref: '../../../beacon-node-oapi.yaml#/components/schemas/Attestation'
+  responses:
+    "200":
+      description: Attestation is stored in pool and broadcast on appropriate subnet
+    "204":
+      description: Attestation is stored in pool and it will be broadcast once all requirements are fulfilled.
```

Comment:
which requirements not be filled but could become filled?

Reply:
```
attestation.data.slot is within the last ATTESTATION_PROPAGATION_SLOT_RANGE slots (within a MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance) -- i.e. attestation.data.slot + ATTESTATION_PROPAGATION_SLOT_RANGE >= current_slot >= attestation.data.slot (a client MAY queue future attestations for processing at the appropriate slot).
```
Also if you received notification that block is received from one node and you submit to another (which hasn't seen that block yet) it will have to wait for that block before validating.


I was reluctant to copy constraints here because it would be hard to keep them up to date if spec changes. Any suggestions how to handle those?

Reply:
should we allow a 204 here? Still seems to put the burden of new functionality on a node (store in pool and query if/when good). It could return an error and allow the message provider to decide what to do

Reply:
I guess it's fine for now. Haven't seen any VC that submits attestation ahead of time